llvm/include/llvm/Passes/StandardInstrumentations.h

```diff
@@ -69,6 +69,62 @@ class PrintIRInstrumentation {
   unsigned CurrentPassNumber = 0;
 };
 
+class DumpIRInstrumentation {
+public:
+  void registerCallbacks(PassInstrumentationCallbacks &PIC);
+
+private:
+  void dumpBeforePass(StringRef PassID, Any IR);
+  void dumpAfterPass(StringRef PassID, Any IR);
```

Comment:
I did not implement a `dumpAfterPassInvalidated` callback, mostly because I didn't understand when that occurs. If it makes sense to include one, I can.